File: app/src/main/java/com/uiic/dwellingpremium/model/Premium.java
package com.uiic.dwellingpremium.model;

import com.uiic.dwellingpremium.utils.DwellingConstants;

import java.io.Serializable;

public class Premium implements Serializable {

    private double sumInsured;
    private int years;
    private double basicPremium;
    private double terrorism;
    private int totalPremium;
    private int serviceTax;
    private int grandTotal;

    public Premium(double sumInsured, int years) {
        this.sumInsured = sumInsured;
        this.years = years;
        calculatePremium();
    }

    public double getSumInsured() {
        return sumInsured;
    }

    public void setSumInsured(double sumInsured) {
        this.sumInsured = sumInsured;
    }

    public int getYears() {
        return years;
    }

    public void setYears(int years) {
        this.years = years;
    }

    public double getBasicPremium() {
        return basicPremium;
    }

    public void setBasicPremium(double basicPremium) {
        this.basicPremium = basicPremium;
    }

    public double getTerrorism() {
        return terrorism;
    }

    public void setTerrorism(double terrorism) {
        this.terrorism = terrorism;
    }

    public int getTotalPremium() {
        return totalPremium;
    }

    public void setTotalPremium(int totalPremium) {
        this.totalPremium = totalPremium;
    }

    public int getServiceTax() {
        return serviceTax;
    }

    public void setServiceTax(int serviceTax) {
        this.serviceTax = serviceTax;
    }

    public int getGrandTotal() {
        return grandTotal;
    }

    public void setGrandTotal(int grandTotal) {
        this.grandTotal = grandTotal;
    }

    private void calculatePremium() {

        this.basicPremium = sumInsured * DwellingConstants.BASIC_PREMIUM_RATE * years;
        this.terrorism = sumInsured * DwellingConstants.TERRORISM * years;
        this.totalPremium = (int) (this.basicPremium + this.terrorism);
        this.serviceTax = (int) (this.totalPremium * DwellingConstants.SERVICE_TAX);
        this.grandTotal = this.totalPremium + this.serviceTax;

    }

}

File: app/src/main/java/com/cyorg/uiic/dwellingpremium/model/Premium.kt
package com.cyorg.uiic.dwellingpremium.model

import android.util.Log
import com.cyorg.uiic.dwellingpremium.utils.DwellingConstants
import java.io.Serializable

/**
 * Created by yorga on 07-07-2017.
 */
data class Premium(var sumInsured: Long,
                   var years: Int,
                   var basicPremium: Long = 0L,
                   var stfi: Long = 0L,
                   var eq: Long = 0L,
                   var discountedBasicPremium: Long = 0L,
                   var totalPremium: Long = 0L,
                   var serviceTax:Long = 0L,
                   var grandTotal: Long = 0L) : Serializable {

    fun calculatePremium(scheme:String = Scheme.Scheme_A.value)  {
        basicPremium = Math.round(sumInsured * DwellingConstants.BASIC_PREMIUM_RATE * years)
        stfi = Math.round(sumInsured * DwellingConstants.STFI_RATE * years)
        eq = Math.round(sumInsured * DwellingConstants.EQ_RATE * years)

        if(scheme == Scheme.Scheme_B.value) {
            if(years < 10)
                discountedBasicPremium = Math.round(basicPremium * ((years * 5)/100.0))
            else
                discountedBasicPremium = Math.round(basicPremium * (50.0/100))
        }

        totalPremium = basicPremium - discountedBasicPremium + stfi + eq
        serviceTax = Math.round(totalPremium * DwellingConstants.SERVICE_TAX)
        grandTotal = totalPremium + serviceTax
    }

}
